dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -553,7 +551,7 @@ public Workflow upsertServiceVersion(@ApiParam(hidden = true) @Auth User user,
         String dockstoreServicePath = upsertVersionHelper(repository, gitReference, null, WorkflowMode.SERVICE, installationAccessToken);
 
         // Add user to service if necessary
-        Workflow service = workflowDAO.findByPath(dockstoreServicePath, false);
+        Workflow service = workflowDAO.findByPath(dockstoreServicePath, false, Service.class).get();
```

Comment:
Maybe this should be in a new class (with the same endpoint path for UI compatibility) for services

Reply:
Do you mean I should have `serviceDAO.findByPath`?

And/or do you mean I should have `ServiceResource.upsertServiceVersion`?

Reply:
The latter